splice_machine/src/main/java/com/splicemachine/derby/vti/ORCRecordIterator.java

```diff
@@ -0,0 +1,246 @@
+	private ExecRow execRow;
+
+	// This is contains the details of the ORC File Record format
+	private StructObjectInspector inspector;
+
+	// Records in Orc File to be processed
+	RecordReader records;
+
+	// Current row in the ORC File to process
+	Object row = null;
+
+	// Indicates if a single file or files in folder are considered
+	boolean isDir = false;
+
+	// Iterator for files in a folder, when folder is specified
+	RemoteIterator<LocatedFileStatus> fileList = null;
+
+	// File System
+	FileSystem filesystem;
+
+	/**
+	 * Constructor performs initialization and sets the record iterator
+	 * 
+	 * @param filesystem
+	 *            : HDFS File System
+	 * @param filePath
+	 *            : File or folder path in HDFS
+	 * @param execRow
+	 *            : Format of the result record
+	 * @throws IOException
+	 */
+	public ORCRecordIterator(FileSystem filesystem, Path filePath,
+			ExecRow execRow) {
+		// set the instance variable of filesystem to use later
+		this.filesystem = filesystem;
+
+		// Set the instance variable of the result record format to be used
+		// later
+		this.execRow = execRow;
+
+		try {
+			Path curFiletoProcess = null;
+			Reader reader;
+
+			// Check if filePath specifes afile or folder
+			// If its folder, set the flag, and get the first file in the folder
+			if (filesystem.isDirectory(filePath)) {
+				isDir = true;
+				this.fileList = filesystem.listFiles(filePath, false);
+				curFiletoProcess = fileList.next().getPath();
+			} else {
+
+				curFiletoProcess = filePath;
+			}
+
+			// Get the reader for the single file (first file in case of folder)
+			reader = getReader(curFiletoProcess);
+
+			// Get the inspector for the format of the record in the ORC File
+			this.inspector = (StructObjectInspector) reader
+					.getObjectInspector();
+
+			// Retrieve the Records from reader to process
+			records = reader.rows();
+
+		} catch (Exception e) {
+			try {
+				if (records != null)
+					records.close();
+			} catch (Exception cE) {
+				throw new RuntimeException(cE);
+			}
+			throw new RuntimeException(e);
+		}
+
+	}
+
+	@Override
+	public Iterator<LocatedRow> iterator() {
+		return this;
+	}
+
+	/*
+	 * This checks if there are more records to process. First checks if there
+	 * are more records in current file. If no more records, gets the next file
+	 * and checks that for more records.
+	 */
+	@Override
+	public boolean hasNext() {
+		try {
+
+			// If no more records in current file
+			if (!records.hasNext()) {
+				// check if there is next file
+				if (this.hasNextFile()) {
+					// Next file exists, so get the records from that file
+					getNextFileRecords();
+				}
+			}
+
+			return records.hasNext();
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/*
+	 * This formats the record in the ORC file and return as LocatedRow
+	 */
+	@Override
+	public LocatedRow next() {
+		try {
+			// Return Row Format
+			ExecRow returnRow = execRow.getClone();
```

Comment:
Without Clone, insert with spliceproperties (i.e. to import), resulted in inserting all records with values from last row. So Clone is required.